caddyhttp/proxy/proxy.go

```diff
@@ -197,10 +218,10 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
 		timeout := host.FailTimeout
 		if timeout > 0 {
 			atomic.AddInt32(&host.Fails, 1)
-			go func(host *UpstreamHost, timeout time.Duration) {
+			go func() {
 				time.Sleep(timeout)
 				atomic.AddInt32(&host.Fails, -1)
-			}(host, timeout)
+			}()
```

Comment:
For reference: https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables

> It is also important to note that variables declared within the body of a loop are not shared between iterations, and thus can be used separately in a closure. 

So this should actually work fine, but it is a dangerous portion of code (for instance if the `for` loop is changed to `for host := range upstream.Selector(r) {`)